sympy/core/numbers.py

```diff
@@ -1288,11 +1288,14 @@ def __gt__(self, other):
             other = _sympify(other)
         except SympifyError:
             raise TypeError("Invalid comparison %s > %s" % (self, other))
-        if isinstance(other, NumberSymbol):
+        if other.is_NumberSymbol:
             return other.__lt__(self)
-        if other.is_comparable:
+        if other.is_Rational and not other.is_Integer:
```

Comment:
<self is Float here>

Reply:
I think assigments to self should be avoided.